pkg/helm/helm.go

```diff
@@ -0,0 +1,264 @@
+	env []string
+}
+
+type Options struct {
+	ConfigHome string
+	KubeConfig *rest.Config
+}
+
+func (o *Options) asEnv() ([]string, error) {
+	if o.ConfigHome == "" {
+		var err error
+		o.ConfigHome, err = os.MkdirTemp(os.TempDir(), "go-helm-client")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	kubeConfigPath := "/dev/null"
+	if o.KubeConfig != nil {
+		kubeConfigPath = path.Join(o.ConfigHome, "kubeconfig")
+		if err := kube.WriteToFile(kube.RestToConfig(o.KubeConfig), kubeConfigPath); err != nil {
+			return nil, err
+		}
+	}
+
+	return []string{
+		fmt.Sprintf("KUBECONFIG=%s", kubeConfigPath),
+		fmt.Sprintf("HELM_CONFIG_HOME=%s", path.Join(o.ConfigHome, "helm-config")),
+	}, nil
+}
+
+// New creates a new helm client.
+func New(opts Options) (*Client, error) {
+	// Clone the host environment.
+	env, err := opts.asEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		env: append(os.Environ(), env...),
+	}, nil
+}
+
+type InstallOptions struct {
+	CreateNamespace bool
+	Name            string
+	Namespace       string
+	Values          map[string]any
+	Version         string
+	NoWait          bool
+	NoWaitForJobs   bool
+}
+
+func (o *InstallOptions) asFlags() ([]string, error) {
+	valuesFile, err := os.CreateTemp(os.TempDir(), "helm-values")
+	if err != nil {
+		return nil, err
+	}
+
+	valuesBytes, err := yaml.Marshal(o.Values)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := valuesFile.Write(valuesBytes); err != nil {
+		return nil, err
+	}
+
+	if err := valuesFile.Close(); err != nil {
+		return nil, err
+	}
+
+	flags := []string{
+		fmt.Sprintf("--namespace=%s", o.Namespace),
+		fmt.Sprintf("--values=%s", valuesFile.Name()),
+	}
+
+	if o.CreateNamespace {
+		flags = append(flags, "--create-namespace")
+	}
+
+	if o.Name == "" {
+		flags = append(flags, "--generate-name")
+	}
+
+	if o.Version != "" {
+		flags = append(flags, fmt.Sprintf("--version=%s", o.Version))
+	}
+
+	if !o.NoWait {
+		flags = append(flags, "--wait")
+	}
+
+	if !o.NoWaitForJobs {
+		flags = append(flags, "--wait-for-jobs")
+	}
+
+	return flags, nil
+}
+
+func (c *Client) List(ctx context.Context) ([]Release, error) {
+	stdout, _, err := c.runHelm(ctx, "list", "-A", "--output=json")
+	if err != nil {
+		return nil, err
+	}
+
+	var releases []Release
+	if err := json.Unmarshal(stdout, &releases); err != nil {
+		return nil, err
+	}
+	return releases, nil
+}
+
+func (c *Client) Install(ctx context.Context, chart string, opts InstallOptions) error {
+	flags, err := opts.asFlags()
+	if err != nil {
+		return err
+	}
+
+	args := []string{"install", chart, "--output=json"}
+	args = append(args, flags...)
+
+	if opts.Name != "" {
+		args = slices.Insert(args, 1, opts.Name)
+	}
+
+	stdout, _, err := c.runHelm(ctx, args...)
+	if err != nil {
+		return err
+	}
+
+	var result map[string]any
+	if err := json.Unmarshal([]byte(stdout), &result); err != nil {
+		return err
+	}
+
+	// TODO: Do somethign with the results
+	fmt.Printf("Result:\n%#v\n", result)
```

Comment:
NIT: Totally agree. The output is currently unreadable with `go test -v ./...`. I would drop it for the moment.

Reply:
Removed for now!